packages/enzyme/src/ReactWrapper.js

```diff
@@ -213,13 +213,13 @@ class ReactWrapper {
    * Forces a re-render. Useful to run before checking the render output if something external
    * may be updating the state of the component somewhere.
    *
-   * NOTE: can only be called on a wrapper instance that is also the root instance.
+   * NOTE: no matter what instance this is called on, it will always update the root.
    *
    * @returns {ReactWrapper}
    */
   update() {
     if (this[ROOT] !== this) {
-      throw new Error('ReactWrapper::update() can only be called on the root');
+      return this[ROOT].update();
```

Comment:
this is the part i want review on, in particular - is this the best way to solve it? should i instead keep the throw, but explicitly `this[ROOT].update()` in most of the places that calls this?

Reply:
Personally I think it makes sense to always update root. I dont see why anyone adding an `update` call that might happen in a non root node not wanting it not to update the root. We do already have a call to `this[ROOT].update()` in `simulate`, if we decide to change this we should update there.

It might be more clear setting `const root = ...` and operating on that instead of calling `this[ROOT].update()` though? When reading a function that calls itself I tend to assume it has n levels of recursion, instead of just one fall through, ie I would assume `this[ROOT]` might itself have a different root.

As an aside, it seems like a lot of things are guarded with `if (this[ROOT] !== this)`, would it make sense to make a `rootOnly(` similar to `single(`?

Reply:
That's a great idea; i'll look into doing that separately.